video/out/opengl/ra.h

```diff
@@ -118,6 +115,18 @@ struct ra_tex {
     void *priv;
 };
 
+struct ra_tex_upload_params {
+    struct ra_tex *tex; // Texture to upload to
+    struct ra_buf *buf; // Buffer to upload from. May be NULL, but only if
+                        // RA_CAP_DIRECT_UPLOAD is set.
+    const void *src;    // Address of data. If `buf` is set, then this is
+                        // interpreted as an offset inside the buffer.
```

Comment:
It's not mirroring the GL API, also mirroring the vulkan API. Not all buffers are host-mapped.